modules/core/src/effects/post-process-effect.ts

```diff
@@ -31,16 +31,23 @@ export default class PostProcessEffect implements Effect {
 
     const {target} = params;
     let inputBuffer = params.inputBuffer;
-    let outputBuffer = params.swapBuffer;
+    let outputBuffer: Framebuffer | null = params.swapBuffer;
 
     for (let index = 0; index < this.passes.length; index++) {
-      if (target && index === this.passes.length - 1) {
+      const isLastPass = index === this.passes.length - 1;
+      const renderToTarget = target !== undefined && isLastPass;
+      if (renderToTarget) {
         outputBuffer = target;
       }
```

Comment:
Question: I'm used to seeing a post-processing stack render into a framebuffer up until the last pass, and then to the drawing buffer. The terminology here suggests the opposite, am I misunderstanding?

Reply:
That is what this is doing. I've tried to make the code clearer, but it is admitedly a bit hard to see what is happening with the buffers being swapped in multiple locations (a single effect can have multiple passes, as well as there being multiple effects applied in sequence)